src/share/routes.js

```diff
@@ -303,6 +331,37 @@ function register(router) {
 
         res.send(note.getContent());
     });
+
+    // Used for searching, require noteId so we know the subTreeRoot
+    router.get('/share/api/search/:noteId', (req, res, next) => {
+        shacaLoader.ensureLoad();
+
+        let note;
+
+        if (!(note = checkNoteAccess(req.params.noteId, req, res))) {
+            return;
+        }
+
+        const {query} = req.query;
```

Comment:
Should `search` be used instead of `query` to better match the ETAPI? Only issue is that the ETAPI uses just `/notes` as an endpoint with a query string, and this one uses `/search/:noteId` since a note id is needed but would overlap with `/notes/:noteId` above.

One possible solution: Switch this endpoint go be `/notes` and require the querystring to include `search` and another parameter such as `context` that provides the current note id as context for the search.

Reply:
Also as per the CodeQL check, this should probably be asserted to not be an array.

Reply:
The current form `/search/:noteId` seems quite cryptic ...

How about: `/notes?query=...&ancestorNoteId=...`?

`ancestorNoteId` defaults to `_share`, but this value is allowed only when there's a `shareIndex` (which basically means that the user wishes that visitors may easily see all shared notes).

Reply:
Where would the client-side JS get the ancestorNoteId/subroot to provide to the API?

Reply:
`subRoot` is given to the template ...

If you want to trigger the search via standard form then something like this:

```html
<form action="notes" method="get">
  <input type="text" name="query">
  <input type="hidden" name="ancestorNoteId" value="<%= subRoot.noteId %>">
</form>
```

If you want to drive this from JS, you can either render it to JS variable or perhaps body element attribute ...

Reply:
Ah okay, just wanted to make sure this wasn't already passed somewhere and I had missed it. Didn't want to go adding anything extra without cause.